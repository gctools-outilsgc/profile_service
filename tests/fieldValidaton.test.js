const mutations = require("../src/resolvers/Mutations");
const { getPrismaTestInstance } = require("./init/prismaTestInstance");

const parent = {};
const ctx = {
    prisma: getPrismaTestInstance()
};

beforeAll(async() => {
    // Create base profile
        const args = {
                gcID: "asdf123456",
                name: "Awesome User",
                email: "awesome.user@somewhere.com"
        };

        const info = "{ gcID, name, email }";
        await mutations.createProfile(parent, args, ctx, info);    

});

afterAll(async() => {
    await getPrismaTestInstance().mutation.deleteProfile({where:{gcID:"asdf123456"}});
});

test("Address field valdiation - streetAddress", async() => {
    const args = {
        gcID:"asdf123456",
        data:{
            address:{
                city:"Ottawa",
                province:"Ontario",
                postalCode:"H0H 0H0",
                country:"Canada"
            }
        }
    };

    const info = "{gcID,name,address{streetAddress, city, province, postalCode, country}";

    await expect(mutations.modifyProfile(parent, args, ctx, info)
    ).rejects.toThrowErrorMatchingSnapshot();
});

test("Address field valdiation - city", async() => {
    const args = {
        gcID:"asdf123456",
        data:{
            address:{
                streetAddress:"450 Blue Nose",
                province:"Ontario",
                postalCode:"H0H 0H0",
                country:"Canada"
            }
        }
    };

    const info = "{gcID,name,address{streetAddress, city, province, postalCode, country}";

    await expect(mutations.modifyProfile(parent, args, ctx, info)
    ).rejects.toThrowErrorMatchingSnapshot();
});

test("Address field valdiation - province", async() => {
    const args = {
        gcID:"asdf123456",
        data:{
            address:{
                streetAddress:"450 Blue Nose",
                city:"Ottawa",
                postalCode:"H0H 0H0",
                country:"Canada"
            }
        }
    };

    const info = "{gcID,name,address{streetAddress, city, province, postalCode, country}";

    await expect(mutations.modifyProfile(parent, args, ctx, info)
    ).rejects.toThrowErrorMatchingSnapshot();
});

test("Address field valdiation - postal code", async() => {
    const args = {
        gcID:"asdf123456",
        data:{
            address:{
                streetAddress:"450 Blue Nose",
                city:"Ottawa",
                province:"Ontario",
                country:"Canada"
            }
        }
    };

    const info = "{gcID,name,address{streetAddress, city, province, postalCode, country}";

    await expect(mutations.modifyProfile(parent, args, ctx, info)
    ).rejects.toThrowErrorMatchingSnapshot();
});

test("Address field valdiation - country", async() => {
    const args = {
        gcID:"asdf123456",
        data:{
            address:{
                streetAddress:"450 Blue Nose",
                city:"Ottawa",
                province:"Ontario",
                postalCode:"H0H 0H0",
            }
        }
    };

    const info = "{gcID,name,address{streetAddress, city, province, postalCode, country}";

    await expect(mutations.modifyProfile(parent, args, ctx, info)
    ).rejects.toThrowErrorMatchingSnapshot();
});

test("Non-existant profile for Modification", async() => {
    const args = {
        gcID: "098ohinjawef",
        data:{
            name:"Face Plant"
        }
    };

    const info = "{gcID, name}";

    await expect(mutations.modifyProfile(parent, args, ctx, info)
    ).rejects.toThrowErrorMatchingSnapshot();
});

test("Non-existant team for Modifcation", async() => {

    const args = {
        id: "098oihalknfdsaasdf",
        data:{
            nameEN:"Faker"
        }
    };

    const info = "{id, nameEN}";

    await expect(mutations.modifyTeam(parent, args, ctx, info)
    ).rejects.toThrowErrorMatchingSnapshot();

});
