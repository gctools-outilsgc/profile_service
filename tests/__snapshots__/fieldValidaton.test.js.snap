// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address field valdiation - city 1`] = `"Missing Fields Required"`;

exports[`Address field valdiation - country 1`] = `"Missing Fields Required"`;

exports[`Address field valdiation - postal code 1`] = `"Missing Fields Required"`;

exports[`Address field valdiation - province 1`] = `"Missing Fields Required"`;

exports[`Address field valdiation - streetAddress 1`] = `"Missing Fields Required"`;

exports[`Non-existant profile for Modification 1`] = `"Profile does not exist"`;

exports[`Non-existant team for Modifcation 1`] = `undefined`;
